src/main/kotlin/org/dashj/platform/dpp/util/Cbor.kt

```diff
@@ -111,18 +113,24 @@ object Cbor {
     fun decodeList(payload: ByteArray): List<Any> {
         try {
             val dataItems = CborDecoder.decode(payload)
-            return readJSONArray(dataItems[0] as co.nstant.`in`.cbor.model.Array).filterNotNull()
+            return readJSONArray(dataItems[0] as Array).filterNotNull()
         } catch (e: ClassCastException) {
             throw CborDecodeException("payload is not a List<Any>", e)
         }
     }
 
-    private fun writeJSONObject(
+    private fun writeRootJSONObject(
         obj: Map<String, Any?>,
-        mapBuilder: MapBuilder<CborBuilder>,
-        baos: ByteArrayOutputStream,
-        innerMapBuilder: AbstractBuilder<*>? = null
+        mapBuilder: MapBuilder<CborBuilder>
     ): CborBuilder {
+        writeJSONObject(obj, mapBuilder)
+        return mapBuilder.end()
+    }
+
+    private fun writeJSONObject(
+        obj: Map<String, Any?>,
+        builder: MapBuilder<*>
+    ) {
```

Comment:
two parameters `baos` and `mapBuilder` were not necessary but were passed to every function.

By removing these, many duplicated functions could be removed.